index.js

```diff
@@ -267,37 +267,47 @@ class ExecutorQueue extends Executor {
      * @async  _start
      * @param  {Object} config               Configuration
      * @param  {Object} [config.annotations] Optional key/value object
+     * @param  {Array}  [config.blockedBy]   Array of job ids that this job is blocked by
      * @param  {String} config.apiUri        Screwdriver's API
+     * @param  {String} config.jobId         JobID that this build belongs to
      * @param  {String} config.buildId       Unique ID for a build
      * @param  {String} config.container     Container for the build to run in
      * @param  {String} config.token         JWT to act on behalf of the build
      * @return {Promise}
      */
     async _start(config) {
         await this.connect();
+        const { buildId, jobId, blockedBy } = config;
 
         // Store the config in redis
         await this.redisBreaker.runCommand('hset', this.buildConfigTable,
-            config.buildId, JSON.stringify(config));
+            buildId, JSON.stringify(config));
+        const startConfig = { buildId, jobId };
+
+        // only include blockedBy key if exists
+        if (blockedBy) {
+            startConfig.blockedBy = blockedBy;
+        }
 
         // Note: arguments to enqueue are [queue name, job name, array of args]
-        return this.queueBreaker.runCommand('enqueue', this.buildQueue, 'start', [{
-            buildId: config.buildId
-        }]);
+        return this.queueBreaker.runCommand('enqueue', this.buildQueue, 'start', [startConfig]);
     }
 
     /**
      * Stop a running or finished build
      * @async  _stop
      * @param  {Object} config               Configuration
      * @param  {String} config.buildId       Unique ID for a build
+     * @param  {String} config.jobId         JobID that this build belongs to
      * @return {Promise}
      */
     async _stop(config) {
         await this.connect();
 
+        // stop does not care about blockedBy
```

Comment:
This won't work if the job has a blockedBy setting, `enqueue` encodes the queue item based on the args passed to it. `del` searches the queue item based on the encoded value of the args.

https://github.com/taskrabbit/node-resque/blob/master/lib/queue.js#L52
https://github.com/taskrabbit/node-resque/blob/master/lib/queue.js#L95

You'll need to pass `blockedBy` to stop() as well